compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -883,7 +883,7 @@ object TypeOps:
           else if symbol.is(Module) then
             TermRef(this(tref.prefix), symbol.sourceModule)
           else if (prefixTVar != null)
-            this(tref)
+            this(tref.applyIfParameterized(tref.typeParams.map(_ => WildcardType)))
```

Comment:
It's not clear why this is correct.

Reply:
I tried to detail it in the commit message (https://github.com/lampepfl/dotty/pull/18706/commits/9460b7d4fb8c2e7ee197831b76d458f2f67cbf71) - given something like:

```scala
class Jacket[T]:
  sealed trait BodyType
```

As we map over the type `Jacket.this.BodyType` we don't want to take the raw type `Jacket`, stripped from its ThisType wrapper, and have it be the prefix of `BodyType`, because we would then get `Jacket#BodyType` which is ill-formed.  Instead we mean to refer to any the BodyType nested class of any Jacket: `Jacket[?]#BodyType`.

Reply:
Thanks for the explanation. I now see the type will be upper bound for `prefixTVar`, so it's harmless.